module3/lessons/week1/HowTheWebWorks.md

```diff
@@ -0,0 +1,110 @@
+> Compare your diagrams in small groups.  What is similar? What is different?  Can you come up with one diagram that best represents your ideas?
+
+<aside class="instructor-notes">
+    <p><strong>Instructor Note</strong><br>Be sure to show an example of a diagram that shows each part of the cycle.</p>
+</aside>
+
+## Request/Response Cycle
+As we start to build out web applications, it is important to be able to visualize the way information flows through the system; typically called the Request/Response Cycle.
+
+First a user gives a client a URL, the client builds a **request** for information (or resources) to be generated by a server.  When the server receives that request, it uses the information included in the request to build a **response** that contains the requested information. Once built, that response is sent back to the client in the requested format, to be rendered to the user.
+
+It is our job as web developers to build out and maintain servers that can successfully build responses based on standardized requests that will be received.  But, what does a standard request look like?  We need to know that before we can start building servers that will respond successfully.
+
+The standard, or protocol we use is **HTTP**.
+
+## HTTP Requests and Responses
+
+The HyperText Transfer Protocol gives us rules about how messages should be sent around the Internet. The system that initiates a connection sends a "request", and the system the answers sends a "response".
+
+### HTTP Request
+
+When a "client" (like a web browser) retrieves information, it sends a payload of data to a server as a "request". This request has many parts, but for now we are going to focus on the **verb** and **path**.
+
+### Verb and Path
+
+Every request needs to be able to tell a server *what* information is requested and *how* that information is being requested.  The *what* is the **path** (also know as a URI), indicating what resource this request is referencing.
+
+Examples of a path:
+
+* `/tasks`
+* `/tasks/4`
+* `/items/6/reviews`
+
+The *how* is the **verb**, indicating what actions the server should take regarding the requested resource.  While the path can vary greatly based on the application, the verbs follow common patterns.  There are 5 common HTTP verbs:
+
+- GET - retrieve some information to be READ by the client/user
+- POST - CREATE a new resource with information contained in the request
+- PUT - UPDATE an entire resource with information contained in the request
+- PATCH - UPDATE a part of a resource with information contained in the request
+- DELETE - DESTROY a resource, typically indicating that it is removed from the database
+
+With these 5 verbs, we send requests that allow us to perform all CRUD functions (create, read, update, destroy) for resources in a database!
+
+> With your small groups,
+> Upate your diagrams to include as much of the request/response header information as you can!
+
```

Comment:
It feels very fitting to me to add a section here where we look at real requests in the browser and see the request and response in dev tools. I have also done an activity where students go to websites they regularly visit and look for requests, then I call on students to share their screen and point out the 1. path, 2. Request Type, 3. Is there a request body? 4. Is there a response body? What is the status code? Or something similar to that.